File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0;
        int maximum = INT_MIN;
        for(auto x : nums){
            sum += x;
            if(sum > maximum) maximum = sum;
            if(sum < 0) sum = 0;
        }
        return maximum;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        maximum = -sys.maxsize
        sum = 0
        
        for i in range(len(nums)):
            sum += nums[i]
            maximum = max(sum, maximum)
            if sum < 0:
                sum = 0
                
        return maximum